### PR TITLE
SCRUM-355 45y

### PR DESCRIPTION
**Pull request by WhiteSenat0r**
Implementation of SCRUM-355

Add meaningful integration test coverage for VehicleModel command behavior.
Focus on tests that validate edge cases around CreateVehicleModelCommandRequest and UpdateVehicleModelCommandRequest using the existing integration testing infrastructure. Follow the style already used in tests/IntegrationTests/VehicleModels/VehicleModelCommandsIntegrationTests.cs.
Requirements:
Do not introduce mocks or an in-memory database.
Use the existing IntegrationTestBase, IntegrationTestsWebApplicationFactory, MediatR Sender, EF Core Context, xUnit, and FluentAssertions patterns.
Add at least 3 new integration tests.
Cover realistic domain/application edge cases, not only empty Guid validation.
Verify both Result state and persisted database state where relevant.
Prefer reusing seeded reference data from the real test database.
Keep the tests deterministic and independent from execution order.
Do not modify production code unless a real defect is discovered and the fix is necessary for the tests to pass.
Preserve the project’s current naming/style conventions.
Suggested cases to consider:
creating a vehicle model with non-existing brand/type IDs should fail and should not persist a new VehicleModel;
creating a duplicate vehicle model name for the same brand should fail or be handled consistently with the existing domain rules;
updating a model with non-existing related type IDs should fail without partially changing existing relationships;
updating available engine/transmission/body/drivetrain types should replace or preserve relationships according to the existing handler behavior.
After implementation, briefly explain what tests were added and why they improve coverage.

# Implementation Plan

## Conceptual Checklist

- Analyze existing test structure and helper method patterns
- Identify seeded reference data available for test scenarios
- Map suggested test cases to validator/handler behavior
- Design test data that ensures deterministic, independent execution
- Verify both Result failures and database non-persistence for negative cases
- Follow existing naming conventions and assertion patterns
- Prioritize edge cases beyond simple empty Guid validation

## Overview

Add at least 3 integration tests to `VehicleModelCommandsIntegrationTests.cs` covering realistic edge cases for CreateVehicleModelCommandRequest and UpdateVehicleModelCommandRequest. Tests will validate domain rules around foreign key constraints, unique constraints, and relationship updates using the real test database with seeded data.

## Assumptions and Rules

From task requirements:
- No mocks or in-memory database
- Use existing IntegrationTestBase, MediatR Sender, EF Core Context, xUnit, FluentAssertions
- Verify both Result state and persisted database state
- Reuse seeded reference data from real test database
- Keep tests deterministic and order-independent
- Only modify production code if real defect discovered
- Preserve project naming/style conventions

From codebase analysis:
- VehicleModel has unique constraint on (Name, VehicleBrandId)
- Validators check existence of Brand/Type/AspectType IDs before persistence
- UpdateVehicleModelCommand replaces aspect collections entirely
- SaveChanges occurs after command execution, enabling rollback on validation failure

## Step-by-Step Implementation

1. **Add test for creating vehicle model with non-existing BrandId**
- Tools: Read existing test structure, Context (for verification), Sender
- Create request with Guid.NewGuid() as BrandId (guaranteed not to exist)
- Assert Result.IsSuccess = false, verify no VehicleModel persisted to database

2. **Add test for creating duplicate model name for same brand**
- Dependencies: Query database for existing brand and model from seeded data
- Create request with same Name + BrandId as existing model
- Assert validation failure due to unique constraint violation
- Verify database count unchanged

3. **Add test for creating model with non-existing EngineType/BodyType IDs**
- Use mix of valid + invalid aspect type IDs (Guid.NewGuid())
- Assert Result failure, verify no partial persistence

4. **Add test for updating model with non-existing aspect type IDs**
- Dependencies: Fetch existing model with GetUpdatedModel() pattern
- Include non-existing aspect type ID in update collection
- Assert failure, verify original aspect collections unchanged

5. **Add test for updating model with non-existing model ID**
- Create UpdateVehicleModelCommandRequest with Guid.NewGuid() as Id
- Assert Result failure with NotFound error

6. **Verify test isolation and determinism**
- Review helper methods for proper data cleanup/isolation
- Ensure tests don't depend on execution order
- Run tests multiple times to confirm stability

## Error Handling and Uncertainties

Potential issues:
- **Race conditions**: Use unique model names per test to avoid conflicts
- **Database state pollution**: Verify IntegrationTestBase properly isolates test scopes
- **Seeded data assumptions**: Confirm seeded brands/types exist before test execution
- **Validator vs DB constraint timing**: Some cases may fail at validation layer vs persistence layer - document which layer catches each edge case in test names

Mitigation: Follow existing test patterns closely, use helper methods for data retrieval, ensure unique test data, verify database state explicitly rather than assuming rollback behavior.

### CODE REVIEW

File: tests/IntegrationTests/VehicleModels/VehicleModelCommandsIntegrationTests.cs [hunk 1]
```diff
@@ -102,14 +102,234 @@ public async Task Send_UpdateRequest_Should_NotUpdateModelIfRequestIsInvalid()
...
+    }
+
+    [Fact]
+    public async Task Send_CreateRequest_ShouldNot_AddNewModelIfTypeIdDoesNotExist()
+    {
+        // Arrange
+        var brand = await Context.Set<VehicleBrand>().FirstAsync();
+        var nonExistingTypeId = Guid.NewGuid();
+        const int minimalProductionYear = 2005;
+        const int maximumProductionYear = 2010;
```
Comment (WhiteSenat0r) [inline tests/IntegrationTests/VehicleModels/VehicleModelCommandsIntegrationTests.cs]: Define reusable const on file level for minimalProductionYear

File: tests/IntegrationTests/VehicleModels/VehicleModelCommandsIntegrationTests.cs [hunk 2]
```diff
@@ -102,14 +102,234 @@ public async Task Send_UpdateRequest_Should_NotUpdateModelIfRequestIsInvalid()
...
+            .Should().Be(initialModelCount);
+    }
+
+    [Fact]
+    public async Task Send_CreateRequest_ShouldNot_AddNewModelIfTypeIdDoesNotExist()
+    {
+        // Arrange
+        var brand = await Context.Set<VehicleBrand>().FirstAsync();
+        var nonExistingTypeId = Guid.NewGuid();
+        const int minimalProductionYear = 2005;
```
Comment (WhiteSenat0r) [inline tests/IntegrationTests/VehicleModels/VehicleModelCommandsIntegrationTests.cs]: do the same thing for maximumProductionYear  as for minimalProductionYear
